Framework/MDAlgorithms/src/CompareMDWorkspaces.cpp

```diff
@@ -31,6 +32,153 @@ class CompareFailsException : public std::runtime_error {
 namespace Mantid {
 namespace MDAlgorithms {
 
+class SimpleMDEvent {
+
+private:
+  std::vector<float> mCoordinates;
+  float mSignal;
+  float mError;
+
+public:
+  SimpleMDEvent(const std::vector<float> &coordinates, const float &signal, const float &error)
+      : mCoordinates(coordinates), mSignal(signal), mError(error) {}
+
+  SimpleMDEvent(const SimpleMDEvent &other)
+      : mCoordinates(other.mCoordinates), mSignal(other.mSignal), mError(other.mError) {}
+
+  // pretty output
+  std::string str() const {
+    std::stringstream ss;
+    size_t numdirs = mCoordinates.size();
+    for (size_t i = 0; i < numdirs; ++i)
+      ss << mCoordinates[i] << ", ";
+    ss << "signal = " << mSignal << ", error = " << mError;
+
+    return ss.str();
+  }
+
+  float getCenter(size_t dim) const { return mCoordinates[dim]; }
+
+  float getSignal() const { return mSignal; }
+
+  float getError() const { return mError; }
+
+  bool operator()(SimpleMDEvent &event1, const SimpleMDEvent &event2) { return event1 < event2; }
+
+  /**
+   * @brief override operator <
+   * order
+   * 1. coordinate
+   * 2. signal
+   * 3. error
+   * @return
+   */
+  bool operator<(const SimpleMDEvent &event2) const {
+
+    bool less(true);
+    bool equal(false);
+
+    // compare coordinates
+    size_t numdirs = mCoordinates.size();
+    for (size_t i = 0; i < numdirs; ++i) {
+      // equal is hard to tell
+      coord_t diff = mCoordinates[i] - event2.mCoordinates[i];
+
+      if (fabs(mCoordinates[i]) > 1E-7 && fabs(diff / mCoordinates[i]) < 1E-6) {
+        // larger than epsilon and relative difference less than epsilon
+        equal = true;
+        less = false;
+      } else if (fabs(mCoordinates[i]) < 1E-7 && diff < 1E-7) {
+        // less than epsilon and absolute difference less than epsilon
+        equal = true;
+        less = false;
+      } else if (diff < 0) {
+        // less
+        less = true;
+      } else {
+        // larger
+        less = false;
+      }
+
+      // no need to continue
+      if (!equal)
+        break;
+    }
+
+    // signal
+    if (equal && mSignal < event2.mSignal) {
+      less = true;
+      equal = false;
+    } else if (equal && mSignal > event2.mSignal) {
+      less = false;
+      equal = false;
+    }
+
+    // error: no need to continue the comparison
+    if (equal && mError >= event2.mError) {
+      less = false;
+    }
+
+    return less;
+  }
+
+  bool operator()(const SimpleMDEvent &lx, const SimpleMDEvent &rx) const { return lx < rx; }
+
+  bool operator>(const SimpleMDEvent &event2) {
```

Comment:
Is this ever used?

Reply:
I don't think so...

Reply:
I would get rid of it. I know it's a pain deleting code after the hard work of writing it but if it's not needed then I think of it as a source of unnecessary complexity.